serverless/cmd/client/client.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package main is a read-only client command for interacting with serverless logs.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/golang/glog"
+	"github.com/google/trillian-examples/serverless/api"
+	"github.com/google/trillian-examples/serverless/internal/client"
+	"github.com/google/trillian/merkle/logverifier"
+	"github.com/google/trillian/merkle/rfc6962/hasher"
+)
+
+var (
+	storageURL   = flag.String("storage_url", "", "Log storage root URL, e.g. file:///path/to/log or https://log.server/and/path")
+	fromIndex    = flag.Uint64("from_index", 0, "Index for inclusion proof")
+	forEntryPath = flag.String("for_entry", "", "Path to entry for inclusion proof")
```

Comment:
There's an odd tension between these flags explicitly laid out for inclusion args, and the setup for a generic tool which takes things via args. I would remove these flags and parse from the args, given how the rest of this code is already shaped up. Or remove the generality stubbed out below, and just do inclusions using these flags.

Reply:
Yeah, it's a bit janky - and tbh the golang Flags stuff doesn't help - looks like it just stops looking for flags once it hits a non-flag arg, so doing `client --logtostderr inclusion --form_index=2 --for_entry=myfile` fails in an unexpected way :/

lemme try your command-with-args approach, I think that might be a lot better